sqlglot/dataframe/sql/dataframe.py

```diff
@@ -0,0 +1,730 @@
+from __future__ import annotations
+
+import functools
+import typing as t
+import zlib
+from copy import copy
+
+import sqlglot
+from sqlglot import expressions as exp
+from sqlglot.dataframe.sql import functions as F
+from sqlglot.dataframe.sql.column import Column
+from sqlglot.dataframe.sql import const
+from sqlglot.dataframe.sql.group import GroupedData
+from sqlglot.dataframe.sql.normalize import normalize
+from sqlglot.dataframe.sql.operations import Operation, operation
+from sqlglot.dataframe.sql.readwriter import DataFrameWriter
+from sqlglot.dataframe.sql.transforms import replace_id_value
+from sqlglot.dataframe.sql.util import get_tables_from_expression_with_join
+from sqlglot.dataframe.sql.window import Window
+from sqlglot.helper import ensure_list
+from sqlglot.optimizer import optimize as optimize_func
+from sqlglot.optimizer.qualify_columns import qualify_columns
+
+if t.TYPE_CHECKING:
+    from sqlglot.dataframe.sql._typing import OutputExpressionContainer
+    from sqlglot.dataframe.sql.session import SparkSession
+
+
+class DataFrame:
+    def __init__(
+        self,
+        spark: SparkSession,
+        expression: exp.Select,
+        branch_id: str = None,
+        sequence_id: str = None,
+        last_op: t.Optional[Operation] = Operation.INIT,
+        pending_hints: t.List[exp.Expression] = None,
+        output_expression_container: OutputExpressionContainer = exp.Select(),
```

Comment:
isn't it dangerous to do this because python evaluates Select() once, it should probably be None and then you do container = container or exp.Select()